test/create_db.sh

```diff
@@ -3,6 +3,37 @@ set -o errexit
 cd $(dirname $0)/..
 source test/db-common.sh
 
+# posix compliant escape sequence
+esc=$'\033'"["
+
+function print_heading() {
+  echo
+  echo -e "${esc}0;34;1m"$1"${esc}0m"
+}
+
+function get_migrations() {
+  migrations=( $(find "$dbpath"/migrations -mindepth 1 -maxdepth 1 -not -path '*/\.*'  2> /dev/null | sort) )
+}
+
+function create_empty_db() {
+  create_script="drop database if exists \`${db}\`; create database if not exists \`${db}\`;"
+  mysql $dbconn -e "$create_script" || die "unable to create ${db}"
+  echo "Created empty "$db" database"
+}
+
+function get_container_version() {
+  container_version="$(goose -path="$dbpath" -env="$dbenv" dbversion | sed 's/goose: dbversion //')"
+}
```

Comment:
I think this doesn't get a container version, but the db version? Which I'm guessing is not actually like "MariaDB 10.3" but is some indication of how far along the migrations have been run?

Also, `get_container_version` and `get_migrations` are both side-effecty. It would be much nicer for them to be "pure" functions that return a value. The way you would do that in Bash is to emit the value on stdout. Then the caller can assign to a variable if it likes.

Reply:
Yeah, bad working on my part. All fixed up now!

Reply:
I also localized the variables and return the outputs in the updated version.